Core/Goals/SkinningGoal.cs

```diff
@@ -70,7 +70,9 @@ public SkinningGoal(ILogger<SkinningGoal> logger, ConfigurableInput input,
         equipmentReader.OnEquipmentChanged += EquipmentReader_OnEquipmentChanged;
 
         //AddPrecondition(GoapKey.dangercombat, false);
-
+        AddPrecondition(GoapKey.pulled, false);
+        AddPrecondition(GoapKey.incombat, false);
+        AddPrecondition(GoapKey.dangercombat, false);
```

Comment:
I would be okay to use `AddPrecondition(GoapKey.dangercombat, false);`
* That would mean that the player killed every known aggroed mob at the time. However anytime it can body pull a new mob.

However `AddPrecondition(GoapKey.incombat, false);` may add many extra seconds of delay due how the server sends out leave combat messages. Sometimes happens in a GCD(1.5s) sometimes takes like 5 seconds.

At this state i would not accept the PR.

Reply:
Sometimes the mob can respawn right next to player while skinning this makes the player incombat; and I saw the bot kept on trying to skin while getting beaten by a mob and died.

Reply:
In case the Agent already entered the `SkinningGoal` it will remain there indefinitely. 

Probably it would be best to refactor the `Update()` loop to don't block the execution.

Reply:
Ah yeah. I can confirm this didn't fix it. I am now trying to add checks for combat inside the whileloop.